lib/logstash/inputs/rabbitmq/march_hare.rb

```diff
@@ -100,6 +102,9 @@ def setup
 
       # exchange binding is optional for the input
       if @exchange
+        @ch.exchange(@exchange,
```

Comment:
This would require a new config option, but I agree it is bad behaviour to
allow it to default (to 'direct'?).
On 29 Jan 2015 3:45 am, "Michael Klishin" <notifications@github.com> wrote:

> In lib/logstash/inputs/rabbitmq/march_hare.rb
> <https://github.com/logstash-plugins/logstash-input-rabbitmq/pull/15#discussion_r23762294>
> :
>
> > @@ -100,6 +102,9 @@ def setup
> >
> >        # exchange binding is optional for the input
> >        if @exchange
> > +        @ch.exchange(@exchange,
>
> Some may find it convenient to have Logstash pre-declare their topology.
> There is no need to declare an exchange to consume from a queue. However,
> if you do, then make sure you provide exchange type, because it defines how
> routing is done and thus is crucially important.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/logstash-plugins/logstash-input-rabbitmq/pull/15/files#r23762294>
> .
>